.golangci.yml

```diff
@@ -58,3 +60,18 @@ issues:
     - linters:
         - lll
       path: "pkg/subctl/operator/common/embeddedyamls/yamls.go"
+
+    # Allow dot imports for `fake` directory and e2e tests
+    - linters:
+        - stylecheck
+      text: "ST1001: "
+      path: _test_suite\.go
```

Comment:
The only file I see that matches this pattern is _pkg/broker/broker_test_suite.go_. You should rename this file properly to _broker_suite_test.go_ instead so you don't need this rule.

Reply:
Not exactly. This is in accordance with [GIngko framework](https://onsi.github.io/ginkgo/#bootstrapping-a-suite). However, there are no specs attached with this suite so basically this file is not needed unless we add any specs to this suite.

Reply:
The file name is not in accordance with the Go test framework nor Gingko, hence the stylecheck error.  There's already a rule (somewhere) that ignores dot import for _*_test.go_  so renaming the file fixes the error. However, since this file isn't really needed, I suggest to remove it - if we ever add unit tests in the future then it can be added with the correct name.

Reply:
That rule to ignore dot imports in _test.go is in `submariner` repo.

For now, I will just remove the file.

Reply:
> That rule to ignore dot imports in _test.go is in `submariner` repo.
> 

hmm... then how are dot imports ignored for the other __test.go_ files in this repo (and in admiral, lighthouse etc)? There must be something in shipyard ...  @dfarrell07 do you know?

Reply:
We have rules like this to ignore dot imports for Go test frameworks:

https://github.com/submariner-io/admiral/blob/devel/.golangci.yml#L69

Reply:
> We have rules like this to ignore dot imports for Go test frameworks:
> 
> https://github.com/submariner-io/admiral/blob/devel/.golangci.yml#L69

ok but I don't see those rules in other projects. Not a big deal - just wondering...

Reply:
I think for any repos without that ignore, it's just that golint and stylecheck are't enabled.

https://github.com/submariner-io/lighthouse/blob/devel/.golangci.yml#L43

Reply:
ahh right

Reply:
But can't we just rename it to suite_test, remove this and move on? 

```
[majopela@bluehat submariner-operator]$ find ./ -name "*suite_test.go"
./pkg/discovery/globalnet/globalnet_suite_test.go
./pkg/discovery/network/network_suite_test.go
./controllers/servicediscovery/servicediscovery_suite_test.go
./controllers/submariner/submariner_suite_test.go
[majopela@bluehat submariner-operator]$ find ./ -name "*test_suite.go"
./pkg/broker/broker_test_suite.go


```

I honestly agree with @tpantelis initial comment, see above ^

Reply:
yeah we can except there's no tests.  If we plan to add them then we can rename and keep it.